### PR TITLE
python3Packages: don't depend on contextlib2

### PR DESCRIPTION
**Pull request by dotlambda**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by dotlambda** [comment]
Result of `nixpkgs-review pr 214700` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>checkov</li>
    <li>emborg (python310Packages.emborg)</li>
    <li>python311Packages.emborg</li>
    <li>python311Packages.quantiphy</li>
  </ul>
</details>
<details>
  <summary>20 packages built:</summary>
  <ul>
    <li>borgbackup</li>
    <li>internetarchive (python310Packages.internetarchive)</li>
    <li>python310Packages.cloudsplaining</li>
    <li>python310Packages.django_silk</li>
    <li>python310Packages.duecredit</li>
    <li>python310Packages.llfuse</li>
    <li>python310Packages.parametrize-from-file</li>
    <li>python310Packages.policy-sentry</li>
    <li>python310Packages.quantiphy</li>
    <li>python310Packages.schema</li>
    <li>python310Packages.tubeup</li>
    <li>python311Packages.cloudsplaining</li>
    <li>python311Packages.django_silk</li>
    <li>python311Packages.duecredit</li>
    <li>python311Packages.internetarchive</li>
    <li>python311Packages.llfuse</li>
    <li>python311Packages.parametrize-from-file</li>
    <li>python311Packages.policy-sentry</li>
    <li>python311Packages.schema</li>
    <li>python311Packages.tubeup</li>
  </ul>
</details>

File: pkgs/development/python-modules/schema/default.nix [hunk 1]
```diff
@@ -19,8 +19,12 @@ buildPythonPackage rec {
...
   };
 
-  propagatedBuildInputs = [
-    contextlib2
+  nativeBuildInputs = [
+    pythonRelaxDepsHook
+  ];
+
+  pythonRemoveDeps = [
+    "contextlib2"
```
Comment (tobim) [inline pkgs/development/python-modules/schema/default.nix:27]:
```suggestion
    # contextlib2 is only needed for backwards compatibility with Python < 3.6,
    # which is not relevant for nixpkgs.
    # Drop after https://github.com/keleshev/schema/pull/273 is released.
    "contextlib2"
```

**Review by tobim — APPROVED** [review]
Fine but please document what this does.

**Comment by dotlambda** [comment]
Result of `nixpkgs-review pr 214700` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>checkov</li>
    <li>python311Packages.cloudsplaining</li>
    <li>python311Packages.emborg</li>
    <li>python311Packages.quantiphy</li>
  </ul>
</details>
<details>
  <summary>20 packages built:</summary>
  <ul>
    <li>datalad</li>
    <li>emborg (python310Packages.emborg)</li>
    <li>internetarchive (python310Packages.internetarchive)</li>
    <li>python310Packages.cloudsplaining</li>
    <li>python310Packages.django_silk</li>
    <li>python310Packages.duecredit</li>
    <li>python310Packages.llfuse</li>
    <li>python310Packages.parametrize-from-file</li>
    <li>python310Packages.policy-sentry</li>
    <li>python310Packages.quantiphy</li>
    <li>python310Packages.schema</li>
    <li>python310Packages.tubeup</li>
    <li>python311Packages.django_silk</li>
    <li>python311Packages.duecredit</li>
    <li>python311Packages.internetarchive</li>
    <li>python311Packages.llfuse</li>
    <li>python311Packages.parametrize-from-file</li>
    <li>python311Packages.policy-sentry</li>
    <li>python311Packages.schema</li>
    <li>python311Packages.tubeup</li>
  </ul>
</details>

**Review by risicle — APPROVED** [review]
`nixpkgs-review` reveals no new failures, macos 10.15 & nixos x86_64

**Comment by mweinelt** [comment]
Result of `nixpkgs-review pr 214700` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages failed to build:</summary>
  <ul>
    <li>checkov</li>
    <li>python311Packages.emborg</li>
    <li>python311Packages.quantiphy</li>
  </ul>
</details>
<details>
  <summary>21 packages built:</summary>
  <ul>
    <li>datalad</li>
    <li>emborg (python310Packages.emborg)</li>
    <li>internetarchive (python310Packages.internetarchive)</li>
    <li>python310Packages.cloudsplaining</li>
    <li>python310Packages.django_silk</li>
    <li>python310Packages.duecredit</li>
    <li>python310Packages.llfuse</li>
    <li>python310Packages.parametrize-from-file</li>
    <li>python310Packages.policy-sentry</li>
    <li>python310Packages.quantiphy</li>
    <li>python310Packages.schema</li>
    <li>python310Packages.tubeup</li>
    <li>python311Packages.cloudsplaining</li>
    <li>python311Packages.django_silk</li>
    <li>python311Packages.duecredit</li>
    <li>python311Packages.internetarchive</li>
    <li>python311Packages.llfuse</li>
    <li>python311Packages.parametrize-from-file</li>
    <li>python311Packages.policy-sentry</li>
    <li>python311Packages.schema</li>
    <li>python311Packages.tubeup</li>
  </ul>
</details>

File: pkgs/development/python-modules/schema/default.nix (hunk 1 above)
Comment (dotlambda) [reply pkgs/development/python-modules/schema/default.nix:27]: Feel free to make a PR.

**Pull request merged**